tests/unit/document/test_multi_modal.py

```diff
@@ -629,3 +629,74 @@ class MMDoc:
     m1 = MMDoc()
     m2 = MMDoc(Document(m1))
     assert m1 == m2
+
+
+def test_acess_multimodal():
+    MyText = TypeVar('MyText', bound=str)
+
+    def my_setter(value) -> 'Document':
+        return Document(text=value + ' but custom!')
+
+    def my_getter(doc: 'Document'):
+        return doc.text
+
+    @dataclass
+    class MyMultiModalDoc:
+        avatar: Image
+        description: Text
+        heading_list: List[Text]
+        heading: MyText = field(setter=my_setter, getter=my_getter, default='')
+
+    m = MyMultiModalDoc(
+        avatar='toydata/test.png',
+        description='hello, world',
+        heading='hello, world',
+        heading_list=['hello', 'world'],
+    )
+    d = Document(m)
```

Comment:
can u add a test after serialization/deserialization in pickle and protobuf and see this still holds.